modules/create_timepoint/jsx/createTimepointIndex.js

```diff
@@ -0,0 +1,362 @@
+  }
+  /**
+   * Retrieve params from the browser URL and save it in state.
+   */
+  collectParams() {
+    const url = new URL(window.location.href);
+    const state = Object.assign({}, this.state);
+    state.url.params = {
+      candID: url.searchParams.get('candID'),
+      identifier: url.searchParams.get('identifier'),
+    };
+    state.data.dccid = state.url.params.candID;
+    this.setState(state);
+  }
+  /**
+   * Used with sending POST data to the server.
+   * @param {object} json - json object converted for POST.
+   * @return {string} send in POST to server.
+   */
+  urlSearchParams(json) {
+    return Object.keys(json).map((key) => {
+      return encodeURIComponent(key) + '=' + encodeURIComponent(json[key]);
+    }).join('&');
+  }
+  /**
+   * Retrieve data from the provided URL and save it in state.
+   */
+  fetchInitializerData() {
+    const state = Object.assign({}, this.state);
+    const send = this.urlSearchParams({
+      command: 'initialize',
+      candID: state.url.params.candID,
+      identifier: state.url.params.identifier,
+      subprojectID: state.form.subproject,
+    });
+
+    fetch(
+      this.props.dataURL, {
+        method: 'POST',
+        mode: 'same-origin',
+        credentials: 'include',
+        redirect: 'follow',
+        headers: {
+          'Accept': 'application/json, text/plain, */*',
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: send,
+      }
+    ).then((response) => response.json())
+      .then(
+        (data) => {
+          // Populate the form errors.
+          if (data.errors) {
+            const state = Object.assign({}, this.state);
+            state.errors = data.errors;
+            this.setState(state);
+          }
+          // Populate the select options for subproject.
+          if (data.subproject) {
+            const state = Object.assign({}, this.state);
+            state.form.options.subproject = data.subproject;
+            state.form.value.subproject = null;
+            state.form.display.subproject = true;
+            this.setState(state);
+          }
+          // Populate the select options for psc.
+          if (data.psc) {
+            const state = Object.assign({}, this.state);
+            state.form.options.psc = data.psc;
+            state.form.value.psc = null;
+            state.form.display.psc = true;
+            this.setState(state);
+          }
+          // Populate the select options for visit.
+          if (data.visit) {
+            // Store the (complete) visit selection information.
+            const state = Object.assign({}, this.state);
+            state.storage.visit = data.visit;
+            this.setState(state);
+            // Handle visit selection.
+            this.handleVisitLabel();
+          }
+          // Display form to user.
+          this.setState({isLoaded: true});
+        }).catch((error) => {
+      this.populateErrors({message: 'Server error.'});
```

Comment:
i think you can remove the call to this function now

Reply:
@zaliqarosli I think it's good practice to have an error to display to the user by the rare chance the server responds with an error. Even if that case should only arise if the server or database goes down at the very moment a post is being sent. Thoughts?

Reply:
@zaliqarosli, I made a change that should resolved it.